test/code/plugins/npmd_config_lib_test.rb

```diff
@@ -16,57 +16,61 @@ class << self
-        @string_agent_config11='<Configuration><Agents><Agent Name="1" Capabilities="1"><IPConfiguration IP="65.171.126.72" Mask="255.255.0.0"/><IPConfiguration IP="65.171.126.73" Mask="255.255.0.0"/></Agent><Agent Name="2" Capabilities="1"><IPConfiguration IP="65.271.126.72" Mask="255.255.248.0"/></Agent></Agents><Networks><Network Name="Default"><Subnet ID="65.171.0.0" Disabled="False" Tag=""/><Subnet ID="198.165.0.0" Disabled="False" Tag=""/></Network><Network Name="NewOne"><Subnet ID="162.128.0.0" Disabled="False" Tag=""/></Network></Networks><Rules><Rule Name="Default" Description="" Protocol="1"><AlertConfiguration><Loss Threshold="-1.0"/><Latency Threshold="-1.0"/></AlertConfiguration><NetworkTestMatrix><SubnetPair SourceSubnet="*" SourceNetwork="*" DestSubnet="*" DestNetwork="*"/></NetworkTestMatrix><Exceptions /></Rule><Rule Name="RuleWithExcept" Description="" Protocol="2"><AlertConfiguration><Loss Threshold="-1.0"/><Latency Threshold="-1.0"/></AlertConfiguration><NetworkTestMatrix><SubnetPair SourceSubnet="*" SourceNetwork="NewOne" DestSubnet="*" DestNetwork="Default"/></NetworkTestMatrix><Exceptions><SubnetPair SourceSubnet="*" SourceNetwork="NewOne" DestSubnet="198.165.0.0" DestNetwork="Default"/></Exceptions></Rule></Rules></Configuration>'.gsub(/\s+/,"")
-        @string_agent_config12='<Configuration><Agents><Agent Name="1" Capabilities="1"><IPConfiguration IP="65.171.126.72" Mask="255.255.0.0"/></Agent></Agents><Networks><Network Name="Default"><Subnet ID="65.171.0.0" Disabled="False" Tag=""/><Subnet ID="198.165.0.0" Disabled="False" Tag=""/></Network></Networks><Rules><Rule Name="Default" Description="" Protocol="1"><AlertConfiguration><Loss Threshold="-1.0"/><Latency Threshold="-1.0"/></AlertConfiguration><NetworkTestMatrix><SubnetPair SourceSubnet="*" SourceNetwork="*" DestSubnet="*" DestNetwork="*"/></NetworkTestMatrix><Exceptions /></Rule><Rule Name="RuleToPartialDrop" Description="" Protocol="1"><AlertConfiguration><Loss Threshold="-1.0"/><Latency Threshold="-1.0"/></AlertConfiguration><NetworkTestMatrix><SubnetPair SourceSubnet="65.171.0.0" SourceNetwork="Default" DestSubnet="198.165.0.0" DestNetwork="NewOne"/></NetworkTestMatrix><Exceptions /></Rule></Rules></Configuration>'.gsub(/\s+/,"")
+        #@test_desc01 = "Test Case 01: Valid UI XML Configuration with default rule"
+        #@test_desc02 = "Test Case 02: Invalid configuration version"
+        #@test_desc03 = "Test Case 03: Mismatched XML tags"
+        #@test_desc04 = "Test Case 04: Invalid configuration tags"
+        #@test_desc05 = "Test Case 05: Invalid json between tags"
+        #@test_desc06 = "Test Case 06: Missing agent capabilities"
+        #@test_desc07 = "Test Case 07: Missing rule protocol"
+        #@test_desc08 = "Test Case 08: Undefined subnet ids in config"
```

Comment:
Lets add a task to write config test cases for both windows and linux with a test plan.

Reply:
Updated all config unit tests